src/explore-education-statistics-common/src/modules/charts/components/MapBlockInternal.tsx

```diff
@@ -726,6 +504,97 @@ export const MapBlockInternal = ({
       </div>
     </>
   );
-};
+}
+
+function generateFeaturesAndDataClasses({
+  classification,
+  dataSetCategories,
+  groups,
+  selectedDataSetConfig,
+}: {
+  classification: DataClassification;
+  dataSetCategories: MapDataSetCategory[];
+  groups: number;
+  selectedDataSetConfig: DataSetCategoryConfig;
+}): {
+  features: MapFeatureCollection;
+  dataClasses: LegendDataClass[];
+} {
+  const selectedDataSetKey = selectedDataSetConfig.dataKey;
+  const {
+    unit,
+    decimalPlaces: dataSetDecimalPlaces,
```

Comment:
I don't think dataSetDecimalPlaces is actually being used if it's set, should it be passed to generateLegendDataClasses?